src/hamcrest/core/core/raises.py

```diff
@@ -41,7 +41,9 @@ def _call_function(self, function: Callable[..., Any]) -> bool:
 
             if isinstance(self.actual, self.expected):
                 if self.pattern is not None:
-                    if re.search(self.pattern, str(self.actual)) is None:
+                    if re.search(
+                        self.pattern, str(self.actual)
+                    ) is None and self.pattern is not str(self.actual):
```

Comment:
`is not` is probably the wrong comparison here; I think you want `!=`

Reply:
fixed